ui/src/shared/utils/fromFlux.test.ts

```diff
@@ -0,0 +1,206 @@
+      '_start',
+      '_stop',
+      '_field',
+      '_measurement',
+      'cpu',
+      'host',
+    ])
+  })
+
+  test('uses the default annotation to fill in empty values', () => {
+    const CSV = `#group,false,false,true,true,true,true
+#datatype,string,long,string,string,long,long
+#default,_result,,,cpu,,6
+,result,table,a,b,c,d
+,,1,usage_guest,,4,
+,,1,usage_guest,,5,`
+
+    const actual = fromFlux(CSV).table
+
+    expect(actual.columns['result'].data).toEqual(['_result', '_result'])
+    expect(actual.columns['a'].data).toEqual(['usage_guest', 'usage_guest'])
+    expect(actual.columns['b'].data).toEqual(['cpu', 'cpu'])
+    expect(actual.columns['c'].data).toEqual([4, 5])
+    expect(actual.columns['d'].data).toEqual([6, 6])
+  })
+
+  test('returns a group key union', () => {
+    const CSV = `#group,true,false,false,true
+#datatype,string,string,string,string
+#default,,,,
+,a,b,c,d
+
+#group,false,false,true,false
+#datatype,string,string,string,string
+#default,,,,
+,a,b,c,d`
+
+    const {fluxGroupKeyUnion} = fromFlux(CSV)
+
+    expect(fluxGroupKeyUnion).toEqual(['a', 'c', 'd'])
+  })
+
+  test('parses empty numeric values as null', () => {
+    const CSV = `#group,false,false,true,true,false,true,true,true,true,true
+#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string,string,string,string
+#default,_result,,,,,,,,,
+,result,table,_start,_stop,_time,_value,_field,_measurement,cpu,host
+,,0,2019-02-01T23:38:32.524234Z,2019-02-01T23:39:02.524234Z,2019-02-01T23:38:33Z,10,usage_guest,cpu,cpu-total,oox4k.local
+,,1,2019-02-01T23:38:32.524234Z,2019-02-01T23:39:02.524234Z,2019-02-01T23:38:43Z,,usage_guest,cpu,cpu-total,oox4k.local`
+
+    const {table} = fromFlux(CSV)
+
+    expect(table.columns['_value'].data).toEqual([10, null])
+  })
+
+  test('handles newlines inside string values', () => {
+    const CSV = `#group,false,false,false,false
+#datatype,string,long,string,long
+#default,_result,,,
+,result,table,message,value
+,,0,howdy,5
+,,0,"hello
+
+there",5
+,,0,hi,6
+
+#group,false,false,false,false
+#datatype,string,long,string,long
+#default,_result,,,
+,result,table,message,value
+,,1,howdy,5
+,,1,"hello
+
+there",5
+,,1,hi,6`
+
+    const {table} = fromFlux(CSV)
+
+    expect(table.columns['value'].data).toEqual([5, 5, 6, 5, 5, 6])
+
+    expect(table.columns['message'].data).toEqual([
+      'howdy',
+      'hello\n\nthere',
+      'hi',
+      'howdy',
+      'hello\n\nthere',
+      'hi',
+    ])
+  })
+
+  test('handles errors that show up on any page', () => {
+    const CSV = `#group,true,false,false,true
+#datatype,string,string,string,string
+#default,,,,
+,a,b,c,d
+
+#group,false,false,true,false
+#datatype,string,string,string,string
+#default,,,,
+,error,reference
+,query terminated: reached maximum allowed memory limits,576`
+
+    expect(() => fromFlux(CSV)).toThrowError(
+      '[576] query terminated: reached maximum allowed memory limits'
+    )
+  })
+})
```

Comment:
Yo dawg, I heard you like tests. Can we get a test to see what happens if the input is defined || `''`